### PR TITLE
[PYTHON-1117] Call ConnectionException with correct kwargs

### PR DESCRIPTION
**Pull request by TheRealFalcon**
https://datastax-oss.atlassian.net/browse/PYTHON-1117

Would an additional test add any value here?

### CODE REVIEW

**Comment by aboudreault** [comment]
Maybe we could add small auth fail test to ensure that this code path is tested. Not sure why we don't have already one.. it would have caught this issue.

**Comment by aboudreault** [comment]
Is your test different than the ones we find in test_authentication.py? What I had in mind was more just a new test in test_authentication that try to connect to one node with bad credentials... and we assert that an AuthenticationFailed exception is raised.

**Comment by TheRealFalcon** [comment]
Yeah, sorry, was waiting for the tests to pass before having you take a look again. This particular code path only gets touched if there's one node that has different auth from the rest (e.g., node1 and node2 allow all, but node3 has password auth), and I didn't see any pre-existing tests that had that.

I put it in a separate file because the other one has a module setup, and it seemed easier to just create a new file than push down the CCM setup logic into each class.

File: tests/integration/standard/test_authentication_misconfiguration.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+
+class MisconfiguredAuthenticationTests(unittest.TestCase):
+    """ One node (not the contact point) has password auth. The rest of the nodes have no auth """
+    @classmethod
+    def setUpClass(cls):
+        if not USE_CASS_EXTERNAL:
+            ccm_cluster = use_cluster(cls.__name__, [3], start=False)
+            node3 = ccm_cluster.nodes['node3']
+            node3.set_configuration_options(values={
```
Comment (TheRealFalcon) [inline tests/integration/standard/test_authentication_misconfiguration.py]: Notice we're only setting password auth on node3 here

**Comment by aboudreault** [comment]
ah! I see. That explains the issue didn't get caught in existing auth tests. Thanks for the clarification!

**Comment by TheRealFalcon** [comment]
Also, I ran the new test on master and the test failed

**Pull request merged**